### DISCUSSION TITLE
Linkerd with Independent Jaeger Installation

### DISCUSSION THREAD

**Discussion by ggolub** [linkerd/linkerd2#5954] [Q&A]
I'm fairly new to both Linkerd and Jaeger so please bear with me.

We are building terraform modules to install all the components in our system.  I've built one to install linkerd v2.9.4 and another to deploy Jaeger 2.19.1.  I need to now get things working together and am a little confused about what is required.  In addition, I see that there were major changes in linkerd v2.10 which has me even more puzzled.

Questions...

1. Can I add the linkerd annotation (linkerd.io/inject: enabled) to the jaeger namespace/pods to ensure MTLS?  Will that have any bearing on tracing?
2. What does setting tracing.enabled in the linkerd helm chart do?  Does that just instruct linkerd to generate traces or is there more such as deploy the OpenCensus collector and/or Jaeger all-in-one?
3. I want to "bring my own" Jaeger. Do I need to install the linkerd-jaeger helm chart?  I already have the Jaeger collector so don't see why the OpenCensus one would be needed.
4. What helm chart changes are required to use my own Jaeger?  I've seen this "linkerd jaeger install --set collector.jaegerAddr='http://my-jaeger-collector.my-jaeger-ns:14268/api/traces' | kubectl apply -f -" but haven't found how to configure this w/o the CLI.

Any advice is greatly appreciated.

**Comment by adleong** [comment]
Hi @ggolub!

1. The `linkerd.io/inject: enabled` annotation is already present on the linkerd-jaeger namespace, so these pods will be injected with the Linkerd-proxy and thus will have mTLS.
2. This enables tracing of the Linkerd control plane itself.  Setting this is unnecessary unless you want to see traces about the inner workings of the Linkerd control plane.
3. See the "Bring your own Jaeger" section of https://linkerd.io/2.10/tasks/distributed-tracing/.  The short answer is that the OpenCensus collector acts as a translator which adapts the OpenCensus API that the Linkerd-proxy emits to the jaeger API.
4. That's correct.  I'm not sure what you mean by configuring it without the CLI.  The CLI's `install` command emits a Kubernetes manifest by rendering a Helm template.

**Reply by ggolub** [reply]
@adleong - Thanks for the quick reply.

1. I was asking about adding the annotation to the **jaeger** namespace so Jaeger itself would be included in the mesh.  If Jaeger is in the service mesh will that cause any problems for tracing?
2. I deployed with trace enabled and can see that it deploys the jaeger all-in-one pod which I don't want.  So I'll leave that disabled.
3. Still not clear about this.  If I want to use my own Jaeger, I would think I shouldn't deploy the linkerd-jaeger helm chart.  If that's the case, how do I configure linkerd to use my Jaeger?  Or is that only relevant if I want to capture traces from linkerd itself?
4. I did see that section of the documentation you mentioned in #3.  That's where I got that cli command.  What I don't see is how to configure linkerd to use my Jaeger, via helm.  Is it the case that I can only use Jaeger with Linkerd if I deploy the extension and, in that case, only gain traces about Linkerd control plane itself?

It seems that I may be overcomplicating this.  I'm going to try just adding Jaeger to the service mesh and NOT use the linkerd-jaeger extension.
